File: worker/tests/helpers/crypto_cli.py
"""CLI bridge used by cross-language tests (src/lib/crypto.test.ts) to
exercise worker/grading/crypto.py from a separate process.
Run with: python crypto_cli.py <encrypt|decrypt> <value>
"""
import sys
from pathlib import Path

sys.path.insert(0, str(Path(__file__).resolve().parents[2]))  # worker/

from grading.crypto import decrypt_token, encrypt_token  # noqa: E402

if __name__ == "__main__":
    action, value = sys.argv[1], sys.argv[2]
    if action == "encrypt":
        print(encrypt_token(value))
    elif action == "decrypt":
        print(decrypt_token(value))
    else:
        raise SystemExit(f"Unknown action: {action}")

File: scripts/crypto_cli.ts
// CLI bridge used by cross-language tests (worker/tests/test_crypto.py) to
// exercise src/lib/crypto.ts from a separate process without importing
// Next.js build machinery. Run with: tsx scripts/crypto_cli.ts <encrypt|decrypt> <value>
import { encryptToken, decryptToken } from '../src/lib/crypto';

const [action, value] = process.argv.slice(2);

if (action === 'encrypt' && value !== undefined) {
  process.stdout.write(encryptToken(value));
} else if (action === 'decrypt' && value !== undefined) {
  process.stdout.write(decryptToken(value));
} else {
  console.error('Usage: crypto_cli.ts <encrypt|decrypt> <value>');
  process.exit(1);
}
